### PR TITLE
False positive as per #2821

### LINKED ISSUE #2821: False Positive on Nuget package Microsoft.VisualStudio.Web.CodeGeneration.*
False positive on the following Nuget packages reported as `cpe:/a:microsoft:visual_studio_code`:

- [Microsoft.VisualStudio.Web.CodeGeneration](https://www.nuget.org/packages/Microsoft.VisualStudio.Web.CodeGeneration/) 
- [Microsoft.VisualStudio.Web.CodeGeneration.Design](https://www.nuget.org/packages/Microsoft.VisualStudio.Web.CodeGeneration.Design/)
- [Microsoft.VisualStudio.Web.CodeGenerators.Mvc](https://www.nuget.org/packages/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/)

### PR DESCRIPTION
**Pull request by stephenbarry-transunion**
## Fixes Issue #
Fixes #2821 

## Description of Change
Suppress false positive in base

## Have test cases been added to cover the new functionality?

N/A

### CODE REVIEW

File: core/src/main/resources/dependencycheck-base-suppression.xml [hunk 1]
```diff
@@ -735,6 +735,13 @@
         <packageUrl regex="true">^pkg:(generic|nuget)/Microsoft\.AspNetCore\.JsonPatch@.*$</packageUrl>
         <cpe>cpe:/a:json-patch_project:json-patch</cpe>
     </suppress>
+    <suppress base="true">
+        <notes><![CDATA[
+        False positives per issue #2821
+        ]]></notes>
+        <packageUrl regex="true">^pkg:(generic|nuget)/Microsoft\.VisualStudio\.Web\.CodeGenerat...\.*$</packageUrl>
```
Comment (jeremylong) [inline core/src/main/resources/dependencycheck-base-suppression.xml]:
```suggestion
        <packageUrl regex="true">^pkg:(generic|nuget)/Microsoft\.VisualStudio\.Web\.CodeGenerat.*$</packageUrl>
```

**Review by jeremylong — CHANGES_REQUESTED** [review]
Did you test the suppression rule? The regex on the packageUrl doesn't look right to me - see my suggestion.

File: core/src/main/resources/dependencycheck-base-suppression.xml (hunk 1 above)
Comment (stephenbarry-transunion) [reply core/src/main/resources/dependencycheck-base-suppression.xml]: Hi Jeremy, I had not, but I have now and you're correct that my initial regex wasn't working as expected. Your suggestion works as expected, see the attached test report.

image

**Review by jeremylong — APPROVED** [review]

**Comment by jeremylong** [comment]
Thanks for the PR

**Pull request merged**